blog/2024-05-21.md

```diff
@@ -0,0 +1,123 @@
+---
+title: It's gonna be Legen... wait for it...
+authors: [gagdiez]
+slug: yield-resume
+tags: [protocol, updates]
+hide_table_of_contents: true
+---
+
+*dary! Legendary! NEAR protocol is getting updated with the ability to yield and resume computations*
+
+![waiting on a loop](../website/static/docs/blog/legendary.jpg)
+
+<!-- truncate -->
+
+## The problem of waiting
+Currently, smart contracts have no way to wait for an external event to happen. This can be a problem when the contract relies on an external service to provide a result.
+
+We encountered this issue while implementing [Chain Signatures](/concepts/abstraction/chain-signatures), which work by requiring an external service to provide a signature.
+
+Until now, the only workaround has been to make the contract call itself in a loop, checking on each iteration if the result is ready. Each call delays the result by one block (~1 second), allowing the contract to wait almost a minute before running out of gas.
+
+![waiting on a loop](../website/static/docs/blog/contract-wait-loop.png)
+*Until now, contracts had to wait by calling themselves until a external service replies... more often than not the contract will run out of gas waiting*
+
+While this method works, it's far from ideal. It wastes a lot of gas on looping and - more often than not - runs out of gas, forcing the user to retry the transaction.
+
+## Yield and Resume
+Starting from version `1.40` of the protocol, developers can **delay the execution** of a function until certain conditions are met (e.g. an external service provides a result).
+
+This way, instead of the contract calling itself on a loop waiting, the contract can simply **yield** calling the function that gives the result. When an external response is provided, the contract will **resume** and return the result.
+
+![waiting on a yield](../website/static/docs/blog/contract-wait-yield.png)
+*Contracts can now yield the execution of a function until an external service signals that the result is ready*
+
+### What is exactly being yielded?
+It is important to notice that the contract is not **halting** or **blocking** its ability to execute, nor **halting in the middle of a function** to later resume it. 
+
+In the same way that a function can return a promise to call another contract, now a function can return a **yield** to call another function.
+
+Indeed, the contract is not halting, but simply **delaying the execution of a callback** until an external agent signals that it is ok to resume.
```

Comment:
This part is well said. Thanks for explaining these important details in a clear manner.